### DISCUSSION TITLE
--window?

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#15898] [General]
WindowsTerminal 1.17.2305.26001
WindowsTerminal 1.18.2305.26002 (preview)

I have these (found via AppPaths).

```
v:\> which wt
wt is an external : D:\wt\exe\windowsterminal.exe

v:\> which wtp
wtp is an external : D:\wtp\exe\windowsterminal.exe
```

When wt is running, the command `wt -w 0 nt`, issued in a shell in the running instance, creates a new tab in the running instance.

When wtp is running, the command `wtp -w 0 nt`, issued in a shell in the running instance, creates a new instance of wtp.

What's going on there?  If that's WAD, how do I get a new tab in the current instance of wtp, with a wtp command line?

Thanks.

**Comment by zadjii-msft** [comment]
How exactly did you install the Terminal to set up those aliases / shortcuts / whatever those are? 

Is there any elevation (running as admin) involved? (thinking of #9628)

How did you initially launch each of `wt` vs `wtp`? (This is me thinking of #15509)

**Comment by vefatica** [comment]
I just unzipped them to the appropriate places.  The aliases come from the registry's AppPaths key (set up by me and checked by shell).

Yes, elevation is involved. [Does that matter to the `wt/wtp -w 0 nt` commands?]

In the experiments in my last post, wtp was run with the command line `schtasks.exe /run /tn wtelevated` where the task uses the command line  `D:\wtp\exe\windowsterminal.exe` and wt was run with the command line `wt` in a shell in the wtp instance (so wt was also elevated).

When either is started from Explorer (Run as Administrator), it exhibits the new instance behavior.

When either is started from Explorer (not elevated), it exhibits the same instance/new tab behavior.

The oddball seems to be the first of my two original experiments where wt was running elevated but I got the same instance/new tab behavior.

**Comment by vefatica** [comment]
P.S., I rarely run wt.  I mention it to point out a difference in behavior.  What I really want is the same instance/new tab behavior from an elevated wtp.

**Comment by vefatica** [comment]
> P.S., I rarely run wt. I mention it to point out a difference in behavior. What I really want is the same instance/new tab behavior from an elevated wtp.

And I can get that behavior (in a rather bizarre fashion, and without the scheduled task business) ...

start d:\wtp\exe\windowsterminal.exe elevated from explorer (right-click, Run as admin, get instance 1).  In this instance,  `wtp -w 0 nt` gives a new instance (close it).

Back in instance 1, issue `d:\wtp\exe\windowsterminal.exe -w -1`.  I get a new instance (instance 2). 

Whether or not I close instance 1 , in instance 2, `wtp -w 0 nt` gives a new tab in the same instance.

I'm sorry it's complicated.  I'll bet there's ultimately something simple going on.

**Comment by vefatica** [comment]
Here  are two pics of that last experiment.  First, the original wtp (in back), having issued `wtp -w 0 np` with the new wtp (in front).

Second, the same after issuing `wtp -w 0 np` in the new wtp.

See that `wtp -w 0 np` in the first instance started a new instance, and the same command in the new instance opened a new tab.

![image](https://github.com/microsoft/terminal/assets/61856645/60769d2c-9cc0-4e30-959a-6bdfd6fea116)

![image](https://github.com/microsoft/terminal/assets/61856645/4d7f4b6c-39a0-4f14-80dd-3f57d8392d32)

**Comment by vefatica** [comment]
This may be just a rehash of previous observations.

There are several ways to start wt elevated where the started instance **opens yet another instance** when a shell running in it executes `wt -w 0 nt`.  They include.

- start menu ... type wt (it's found) ... run as admin ... yes at UAC prompt
- explorer ... right-click the exe ... run as admin ... yes at UAC prompt
- scheduled task, on-demand andrun with highest privileges ... start the task with schtasks.exe ... avoids the UAC prompt

There's one (rather general) way to start wt elevated where the started instance **opens a new tab** when a shell running in it executes `wt -w 0 nt`.  Namely,

- start windowsterminal.exe (path if needed) from an (any) elevated app.
